libs/go/aea_end2end/main.go

```diff
@@ -0,0 +1,95 @@
+/* -*- coding: utf-8 -*-
+* ------------------------------------------------------------------------------
+*
+*   Copyright 2018-2021 Fetch.AI Limited
+*
+*   Licensed under the Apache License, Version 2.0 (the "License");
+*   you may not use this file except in compliance with the License.
+*   You may obtain a copy of the License at
+*
+*       http://www.apache.org/licenses/LICENSE-2.0
+*
+*   Unless required by applicable law or agreed to in writing, software
+*   distributed under the License is distributed on an "AS IS" BASIS,
+*   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*   See the License for the specific language governing permissions and
+*   limitations under the License.
+*
+* ------------------------------------------------------------------------------
+ */
+package main
+
+import (
+	"log"
+	"os"
+	"os/signal"
+
+	aea "aealite"
+	connections "aealite/connections"
+	protocols "aealite/protocols"
+	helpers "aealite/helpers"
+)
+
+
+func getRole (protocols.ProtocolMessageInterface, protocols.Address) protocols.Role{
+	return protocols.Role("some")
+}
+
+func makeSellerDialogues(address string) * protocols.Dialogues{
+	set := helpers.NewSet()
+    set.Add("Some")
+    dialogues := protocols.NewDialogues(protocols.Address(address), set, getRole, false, "my dialogues")
+    return dialogues
+}
+
+
+func main() {
+
+	var err error
+
+	// env file
+	if len(os.Args) != 2 {
+		log.Print("Usage: main ENV_FILE")
+		os.Exit(1)
+	}
+	envFile := os.Args[1]
+
+	log.Print("Agent starting ...")
+
+
+	// Create agent
+	agent := aea.Agent{}
+
+	// Set connection
+	agent.Connection = &connections.P2PClientApi{}
+
+	// Initialise agent from environment file (first arg to process)
+	err = agent.InitFromEnv(envFile)
+	if err != nil {
+		log.Fatal("Failed to initialise agent", err)
+	}
+	log.Print("successfully initialized AEA!")
+
+	err = agent.Start()
+	if err != nil {
+		log.Fatal("Failed to start agent", err)
+	}
+	log.Print("successfully started AEA!")
+	
+	
+	dialogues := makeSellerDialogues(agent.Address())
+	envelope := agent.Get()
+    log.Printf("got incoming envelope: %s", envelope)
+    dialogues.Update(envelope)
```

Comment:
Then you should continue the dialogue: respond with PROPOSE, receive ACCEPT, send MATCH ACCEPT, etc